packages/widget/src/controllers/transfers/evm/build.ts

```diff
@@ -0,0 +1,46 @@
+import { EVMAssetTransfer } from '@buildwithsygma/sygma-sdk-core';
+import { Web3Provider } from '@ethersproject/providers';
+import { type FungibleTokenTransferController } from '../fungibleTokenTransfer';
+
+/**
+ * @dev If we did proper validation this shouldn't throw.
+ * NBot sure how to handle if it throws :shrug:
+ */
+export async function buildEvmFungibleTransactions(
+  this: FungibleTokenTransferController
+): Promise<void> {
+  //we already check that but this prevents those typescript errors
+  const provider = this.walletContext.value?.evmWallet?.provider;
+  const providerChaiId = this.walletContext.value?.evmWallet?.providerChainId;
+  const address = this.walletContext.value?.evmWallet?.address;
+  if (
+    !this.sourceNetwork ||
+    !this.destinationNetwork ||
+    !this.resourceAmount ||
+    !this.selectedResource ||
+    !this.destinatonAddress ||
+    !provider ||
+    !address ||
+    providerChaiId !== this.sourceNetwork.chainId
+  ) {
+    return;
+  }
+
+  const evmTransfer = new EVMAssetTransfer();
+  await evmTransfer.init(new Web3Provider(provider, providerChaiId), this.env);
+  const transfer = await evmTransfer.createFungibleTransfer(
+    address,
+    this.destinationNetwork.chainId,
+    this.destinatonAddress,
+    this.selectedResource.resourceId,
+    String(this.resourceAmount)
+  );
+  const fee = await evmTransfer.getFee(transfer);
```

Comment:
🤔 nit: so this function not only builds the evm fungible transfer, but also get the fee and build approvals? Does it makes sense to you to separate this into new functions?